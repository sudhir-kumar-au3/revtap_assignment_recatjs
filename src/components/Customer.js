import React, { useState, useEffect } from "react";
import { fetchData } from "../apicall";
import List from "./List";
function Customer() {
  const [customerData, setCustomerData] = useState([]);
  const [limitSize, setLimitSize] = useState(5);

  const onLoadMore = () => {
    setLimitSize(limitSize + 5);
  };
  useEffect(() => {
    fetchData().then((data) => {
      setCustomerData(data.customers);
    });
  }, []);
  return (
    <div className='bg-light m-4'>
        <h5>Customers</h5>
      <table className="table p-2">
        <thead className="thead-light">
          <tr>
            <th>ID</th>
            <th>First Name</th>
            <th>Last Name</th>
            <th>Email</th>
            <th>Created</th>
            <th>Orders</th>
          </tr>
        </thead>
        <tbody>
          {customerData.length > 0
            ? customerData.slice(0, limitSize).map((data) => {
                return (
                  <List
                    key={data.id}
                    id={data.id}
                    firstName={data.firstName}
                    lastName={data.lastName}
                    email={data.email}
                    created={data.created}
                    orders={data.orders}
                  ></List>
                );
              })
            : `Loading...`}
        </tbody>
      </table>
      <div className="text-center">
        <button className="btn btn-info" onClick={onLoadMore}>
          Load More
        </button>
      </div>
    </div>
  );
}

export default Customer;
