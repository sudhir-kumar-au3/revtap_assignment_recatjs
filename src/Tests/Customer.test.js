import React from "react";
import renderer from "react-test-renderer";
import Customer from "../components/Customer";

it("renders customer component correctly", () => {
  const tree = renderer.create(<Customer></Customer>).toJSON();
  expect(tree).toMatchSnapshot();
});