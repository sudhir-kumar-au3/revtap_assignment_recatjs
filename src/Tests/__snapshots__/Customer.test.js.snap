// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="bg-light m-4"
>
  <h5>
    Customers
  </h5>
  <table
    className="table p-2"
  >
    <thead
      className="thead-light"
    >
      <tr>
        <th>
          ID
        </th>
        <th>
          First Name
        </th>
        <th>
          Last Name
        </th>
        <th>
          Email
        </th>
        <th>
          Created
        </th>
        <th>
          Orders
        </th>
      </tr>
    </thead>
    <tbody>
      Loading...
    </tbody>
  </table>
  <div
    className="text-center"
  >
    <button
      className="btn btn-info"
      onClick={[Function]}
    >
      Load More
    </button>
  </div>
</div>
`;

exports[`renders customer component correctly 1`] = `
<div
  className="bg-light m-4"
>
  <h5>
    Customers
  </h5>
  <table
    className="table p-2"
  >
    <thead
      className="thead-light"
    >
      <tr>
        <th>
          ID
        </th>
        <th>
          First Name
        </th>
        <th>
          Last Name
        </th>
        <th>
          Email
        </th>
        <th>
          Created
        </th>
        <th>
          Orders
        </th>
      </tr>
    </thead>
    <tbody>
      Loading...
    </tbody>
  </table>
  <div
    className="text-center"
  >
    <button
      className="btn btn-info"
      onClick={[Function]}
    >
      Load More
    </button>
  </div>
</div>
`;
